clang/include/clang/Basic/SyclOptReportHandler.h

```diff
@@ -28,18 +28,26 @@ class SyclOptReportHandler {
     std::string KernelArgName;
     std::string KernelArgType;
     SourceLocation KernelArgLoc;
+    unsigned KernelArgSize;
+    std::string KernelArgDesc;
+    std::string KernelArgParent;
 
     OptReportInfo(std::string ArgName, std::string ArgType,
-                  SourceLocation ArgLoc)
+                  SourceLocation ArgLoc, unsigned ArgSize, std::string ArgDesc,
+                  std::string ArgParent)
         : KernelArgName(std::move(ArgName)), KernelArgType(std::move(ArgType)),
-          KernelArgLoc(ArgLoc) {}
+          KernelArgLoc(ArgLoc), KernelArgSize(ArgSize),
+          KernelArgDesc(std::move(ArgDesc)),
+          KernelArgParent(std::move(ArgParent)) {}
   };
   llvm::DenseMap<const FunctionDecl *, SmallVector<OptReportInfo>> Map;
 
 public:
   void AddKernelArgs(const FunctionDecl *FD, std::string ArgName,
```

Comment:
I think this should take the parameters via `StringRef` rather than `std::string`.

Reply:
Should I change the types in OptReportInfo as well? I did not because of this comment - https://github.com/intel/llvm/pull/3492#discussion_r615990453

Reply:
Making sure we're on the same page -- I think the fields should remain a `std::string` here, and I am only recommending changing the parameters to be `StringRef` (so the copy will still happen, but there's no need to do the `move` dance and callers don't have to construct the string on their end).